### ISSUE TITLE
Can't add expenses without Full Permission

### ISSUE THREAD

**Issue by alanp5** [dmp1ce/redmine-contracts-with-time-tracking-plugin#67]
I am trying to setup contracts so that users can bill against them and log expenses. So far the billing part is working as expected however, it doesn't seem that there is any way to allow employees to add expenses without giving them full access to the contract. Below are the permissions that I'm using currently which allows employees to select the contract when billing their time. Is there a way something similar can be done for expenses?

![capture](https://user-images.githubusercontent.com/1714538/34496251-d889d096-efc6-11e7-92d4-6fd12aea32b8.PNG)

**Comment by partyp** [comment]
With your current permissions you can go directly here: `/projects/[project_name]/expenses/new` and add an expense. Although, the user will not be able to see that it was posted anywhere because they don't have the "view all contracts for project" and/or "view contract details" permissions.

How would you ideally expect this to work?

**Comment by alanp5** [comment]
Right, I figured that out. Ideally it would work just like the time keeping aspect in that you can add expenses associated with a contract without having the ability to see the details of the contract. The time keeping aspect already works that way but expenses do not.
